go/interpreter/lfvm/interpreter_test.go

```diff
@@ -453,6 +462,427 @@ func TestOKInstructionPath(t *testing.T) {
 	}
 }
 
+func TestRunReturnsEmptyResultOnEmptyCode(t *testing.T) {
+	// Create execution context.
+	ctxt := getEmptyContext()
+	// Get tosca.Parameters
+	ctxt.params = tosca.Parameters{
+		Input:  []byte{},
+		Static: true,
+		Gas:    10,
+	}
+	ctxt.code = make([]Instruction, 0)
+
+	// Run testing code
+	result, err := Run(ctxt.params, ctxt.code, false, true, false)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	if result.Output != nil {
+		t.Errorf("unexpected output: want nil, got %v", result.Output)
+	}
+	if result.GasLeft != ctxt.params.Gas {
+		t.Errorf("unexpected gas left: want %v, got %v", 10, result.GasLeft)
+	}
+	if !result.Success {
+		t.Errorf("unexpected success: want true, got false")
+	}
+}
+
+func TestRunWithStatistics(t *testing.T) {
+	// Create execution context.
+	ctxt := getEmptyContext()
+	// Get tosca.Parameters
+	ctxt.params = tosca.Parameters{
+		Input:  []byte{},
+		Static: true,
+		Gas:    10,
+		Code:   []byte{byte(STOP), 0},
+	}
+	ctxt.code = []Instruction{{STOP, 0}}
+
+	// Run testing code
+	_, err := Run(ctxt.params, ctxt.code, true, true, false)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	if got := global_statistics.single_count[uint64(STOP)]; got != 1 {
+		t.Errorf("unexpected statistics: want 1 stop, got %v", got)
+	}
+}
+
+func TestRunWithLogging(t *testing.T) {
+	// Create execution context.
+	ctxt := getEmptyContext()
+	instructions := []Instruction{
+		{PUSH1, 1},
+		{STOP, 0}}
+
+	// Get tosca.Parameters
+	ctxt.params = tosca.Parameters{
+		Input:  []byte{},
+		Static: true,
+		Gas:    10,
+		Code:   []byte{0x0},
+	}
+	ctxt.code = instructions
+
+	// redirect stdout
+	old := os.Stdout
+	r, w, _ := os.Pipe()
+	os.Stdout = w
```

Comment:
This is fine for now, to increase the test coverage, but in general the logging mode should get an improvement. The logging target should not be `os.Stdout` but some log consumer with some mock-able interface (just fyi, nothing to be changed here).